File: Two Pointers/633_Sum of Square Numbers.cpp
//和167思路类似
class Solution {
public:
    bool judgeSquareSum(int c) {
        int x=0,y=pow(c,0.5)+1;
        while(x<=y){
            auto sum=pow(x,2)+pow(y,2);
            if(sum == c) return true;
            else if (sum <c) x++;
            else y--;
        }
        return false;

    }
};

File: Two Pointers/633_Sum of Square Numbers.py
class Solution:
    def judgeSquareSum(self, c: int) -> bool:
        i=0
        j=pow(c,0.5)+1
        j=int(j)
        while i<=j:
            sum=pow(i,2)+pow(j,2)
            if sum == c:
                return True
            elif sum >c:
                j-=1
            elif sum<c:
                i+=1
        return False
